app/src/main/java/org/mozilla/fenix/share/ShareController.kt

```diff
@@ -73,15 +83,15 @@ class DefaultShareController(
     }
 
     override fun handleShareToDevice(device: Device) {
-        sendTabUseCases.sendToDeviceAsync(device.id, sharedTabs.toTabData())
-        (fragment.activity as ShareFragment.TabsSharedCallback).onTabsShared(sharedTabs.size)
-        dismiss()
+        with(fragment.activity!!.applicationContext) {
```

Comment:
Maybe it's possible to pass in the context of the activity to the controller and use that?

The `!!` assertion has a tendency to bite back with Android lifetimes. I don't believe you'd need the application lifetime to send the tab either.

Reply:
Having just reviewed https://github.com/mozilla-mobile/fenix/pull/5108/files, you should have access to the context from there when it merges since it adds in that change.

Reply:
Thanks!